code/modules/reagents/Chemistry-Recipes/Slime_extracts.dm

```diff
@@ -618,3 +618,21 @@
 	var/obj/P = new chosen
 	if(P)
 		P.loc = get_turf(holder.my_atom)
+
+//Rainbow :o)
+/datum/chemical_reaction/slimeRNG
+	name = "Random Core"
+	id = "slimerng"
+	result = null
+	required_reagents = list("plasma" = 1)
+	result_amount = 1
+	required_other = 1
+	required_container = /obj/item/slime_extract/rainbow
+
+/datum/chemical_reaction/slimeRNG/on_reaction(var/datum/reagents/holder)
+	feedback_add_details("slime_cores_used","[replacetext(name," ","_")]")
+	for(var/mob/O in viewers(get_turf(holder.my_atom), null))
+		O.show_message(text("<span class='danger'>Infused with plasma, the core begins to quiver and grow, and soon a new baby slime emerges from it!</span>"), 1)
```

Comment:
was just keeping form with how other slime cores did it, but yeah I don't see why that wouldn't work.